src/main/java/de/tum/in/www1/artemis/service/ExerciseGroupService.java

```diff
@@ -0,0 +1,107 @@
+        this.examAccessService = examAccessService;
+    }
+
+    /**
+     * Save an exerciseGroup
+     *
+     * @param exerciseGroup the entity to save
+     * @return the persisted entity
+     */
+    public ExerciseGroup save(ExerciseGroup exerciseGroup) {
+        log.debug("Request to save exerciseGroup : {}", exerciseGroup);
+        return exerciseGroupRepository.save(exerciseGroup);
+    }
+
+    /**
+     * Get one exercise group by id.
+     *
+     * @param exerciseGroupId the id of the exercise group
+     * @return the entity
+     */
+    @NotNull
+    public ExerciseGroup findOne(Long exerciseGroupId) {
+        log.debug("Request to get exercise group : {}", exerciseGroupId);
+        return exerciseGroupRepository.findById(exerciseGroupId)
+                .orElseThrow(() -> new EntityNotFoundException("Exercise group with id \"" + exerciseGroupId + "\" does not exist"));
+    }
+
+    /**
+     * Get all exercise groups for the given exam.
+     *
+     * @param examId the id of the exam
+     * @return the list of all exercise groups
+     */
+    public List<ExerciseGroup> findAllByExamId(Long examId) {
+        log.debug("REST request to get all exercise groups for Exam : {}", examId);
+        return exerciseGroupRepository.findByExamId(examId);
+    }
+
+    /**
+     * Delete the exercise group by id.
+     *
+     * @param exerciseGroupId the id of the entity
+     */
+    public void delete(Long exerciseGroupId) {
+        log.debug("Request to delete exercise group : {}", exerciseGroupId);
+        exerciseGroupRepository.deleteById(exerciseGroupId);
+    }
+
+    /**
+     * Checks if the current user is allowed to manage exams of the given course, that the exam exists,
+     * that the exam belongs to the given course and the exercise group belongs to the given exam.
+     *
+     * @param courseId          The id of the course
+     * @param examId            The id of the exam
+     * @param exerciseGroupId   The id of the exercise group
+     * @param <X>               The type of the return type of the requesting route so that the
+     *                          response can be returned there
+     * @return an Optional with a typed ResponseEntity. If it is empty all checks passed
+     */
+    public <X> Optional<ResponseEntity<X>> checkCourseAndExamAndExerciseGroupAccess(Long courseId, Long examId, Long exerciseGroupId) {
+        Optional<ResponseEntity<X>> courseAndExamAccessFailure = examAccessService.checkCourseAndExamAccess(courseId, examId);
+        if (courseAndExamAccessFailure.isPresent()) {
+            return courseAndExamAccessFailure;
+        }
+        Optional<ExerciseGroup> exerciseGroup = exerciseGroupRepository.findById(exerciseGroupId);
+        if (exerciseGroup.isEmpty()) {
+            return Optional.of(notFound());
+        }
+        if (!exerciseGroup.get().getExam().getId().equals(examId)) {
+            return Optional.of(forbidden());
```

Comment:
I don't think forbidden() is a good choice here. The user is allowed however the exam doesn't belong to the exercise group.
You could either agrue with a "notFound()" here since you can't finde an exercise belonging to the course, maybe a HTTP 409 would be a good choice here?